### PR TITLE
Turn on /features:strict for all projects

### LINKED ISSUE #13203: No errors reported when out parameter of ImmutableArray type is not initialized before function returns
in roslyn\src\ExpressionEvaluator\CSharp\Source\ExpressionCompiler\CompilationContext.cs

There is a following code at the end of ExtendBinderChain function, which has out parameter ```out ImmutableArray<LocalSymbol> declaredLocals```

```
            if (declaredLocalsScopeDesignator != null)
            {
                declaredLocals = originalRootBinder.GetDeclaredLocalsForScope(declaredLocalsScopeDesignator);
            }
            else
            {
                declaredLocals = ImmutableArray<LocalSymbol>.Empty;
            }
```

If the ```else``` block is removed, no error reported that declaredLocals might be not fully assigned.

Expected to get an error.

### PR DESCRIPTION
**Pull request by gafter**
This enables us to get better definite assignment behavior for
struct types imported from metadata that have only private fields.
Closes #13203.

@AlekseyTs @dotnet/roslyn-interactive @dotnet/roslyn-compiler Please review

### CODE REVIEW

**Comment by cston** [comment]
LGTM

File: src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Resolution.cs [hunk 1]
```diff
@@ -63,6 +63,8 @@ public ResolvedReference(int index, MetadataImageKind kind)
                 Debug.Assert(index >= 0);
                 _index = index + 1;
                 _kind = kind;
+                _aliasesOpt = ImmutableArray<string>.Empty;
```
Comment (tmat) [inline src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Resolution.cs]: both should be ```default(ImmutableArray<string>)```

**Comment by gafter** [comment]
Should be closes #13223.

**Comment by agocke** [comment]
@gafter I think we should consider turning this on for every project and producing an error if someone tries to turn it off. Similar to how we handle checked/unchecked.

**Comment by jaredpar** [comment]
I agree with @agocke. Let's do this for all projects.  Just need to put the entry here instead:

https://github.com/dotnet/roslyn/blob/master/build/Targets/Settings.targets#L214

**Comment by gafter** [comment]
@jaredpar @agocke Updated as recommended. Please re-review.

**Comment by agocke** [comment]
LGTM, although it looks like there are more places that need fixes.

**Comment by VSadov** [comment]
The ```!= null``` cases that were fixed look like actual bugs!!!

**Comment by VSadov** [comment]
LGTM

**Pull request merged**